### PR TITLE
RFR: Backend GET /books Fuzzy Search

### LINKED ISSUE #102: Fuzzy searching not working for books
Not sure if this is a known bug but the searching isn't functioning as expected—searching "hu" for example should turn up "Huckleberry Finn" but instead shows nothing. But searching for "h" works. 
screen shot 2018-11-09 at 1 02 37 am
screen shot 2018-11-09 at 1 02 34 am

### PR DESCRIPTION
**Pull request by Davidw1339**
## Status: 
:rocket: Ready

## Description
Had to do some digging but found a case insensitive contains basically for fuzzy searching

Related issues: fixes #102

### CODE REVIEW

File: backend/api/views/book.py [hunk 1]
```diff
@@ -150,8 +150,13 @@ def find_books():
     if "search_string" in user_data:
         tokens = user_data["search_string"].split(" ")
         for search in tokens:
+            case_ins_search = "%{0}%".format(
```
Comment (micleo2) [inline backend/api/views/book.py]: :lock: good work

File: backend/api/views/book.py [hunk 2]
```diff
@@ -150,8 +150,13 @@ def find_books():
...
+                search
+            )  # https://stackoverflow.com/questions/4926757/sqlalchemy-query-where-a-column-contains-a-substring
             filtered_books = filtered_books.filter(
-                or_(Book.name.contains(search), Book.author.contains(search))
+                or_(
+                    Book.name.ilike(case_ins_search), Book.author.ilike(case_ins_search)
```
Comment (micleo2) [inline backend/api/views/book.py]: :+1: unexpected change

File: backend/api/views/book.py [hunk 3]
```diff
@@ -150,8 +150,13 @@ def find_books():
...
+                )
```
Comment (micleo2) [inline backend/api/views/book.py]: squash :clap: and merge :twisted_rightwards_arrows:

File: backend/api/views/book.py [hunk 4]
```diff
@@ -150,8 +150,13 @@ def find_books():
     if "search_string" in user_data:
         tokens = user_data["search_string"].split(" ")
         for search in tokens:
+            case_ins_search = "%{0}%".format(
+                search
+            )  # https://stackoverflow.com/questions/4926757/sqlalchemy-query-where-a-column-contains-a-substring
```
Comment (XuanLi-Leong) [inline backend/api/views/book.py]: Great documentation practice ✍️ 📝

**Pull request merged**